app/Http/Controllers/OAuthController.php

```diff
@@ -2,20 +2,78 @@
 
 namespace App\Http\Controllers;
 
-use App\Models\User;
-use CDash\Middleware\OAuth2\OAuth2Interface;
 use Illuminate\Http\RedirectResponse;
 use Illuminate\Http\Request;
 use Illuminate\Support\Collection;
 use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Str;
+use Laravel\Socialite\Facades\Socialite;
+use Symfony\Component\HttpFoundation\RedirectResponse as symfonyResponse;
 use RuntimeException;
 
+use App\Models\User;
+use App\Http\Controllers\Auth\RegisterController;
+use CDash\Middleware\OAuth2\OAuth2Interface;
+
 /**
  * Class OAuthController
  * @package App\Http\Controllers
  */
 final class OAuthController extends AbstractController
 {
+    /**
+     * @param Request $request
+     * @param string $service
+     * @return symfonyResponse
+     */
+
+    public function socialite(Request $request, string $service)
+    {
+        return Socialite::driver($service)->redirect();
+    }
+
+    /**
+     * @param string $service
+     * @param User $user_table
+     * @return RedirectResponse
+    */
+
+    public function callback(string $service, User $user_table)
+    {
+        $authUser = Socialite::driver($service)->user();
+        $email =  $authUser->getEmail();
+        [$fname, $lname] = explode(" ", $authUser->getName());
+
+        // TODO: What if, for whatever reason, there is more than one user found?
+        $user = $user_table->whereIn('email', [$email])->first();
```

Comment:
If you only have one condition, you can use `where()` instead of `whereIn()`.  You can also take advantage of Eloquent's [`firstWhere()`](https://laravel.com/docs/10.x/eloquent#retrieving-single-models) method to simplify it even further.  Since you're just executing a static query, you should also use the static method syntax.

```suggestion
        $user = $user_table::firstWhere('email', $email);
```